apps/api-extractor/src/generators/ApiReportGenerator.ts

```diff
@@ -308,10 +308,13 @@ export class ApiReportGenerator {
 
       case ts.SyntaxKind.SyntaxList:
         if (span.parent) {
-          if (AstDeclaration.isSupportedSyntaxKind(span.parent.kind)) {
-            // If the immediate parent is an API declaration, and the immediate children are API declarations,
-            // then sort the children alphabetically
-            sortChildren = true;
+          if (span.parent.kind === ts.SyntaxKind.EnumDeclaration
+            && collector.extractorConfig.memberSortOrder === EnumMemberOrder.preserve) {
+            //Configured to keep enum member source ordering, leave sortChildren false
+          } else if (AstDeclaration.isSupportedSyntaxKind(span.parent.kind)) {
+              // If the immediate parent is an API declaration, and the immediate children are API declarations,
+              // then sort the children alphabetically
+              sortChildren = true;
```

Comment:
The empty `if` block is a little odd. Potentially consider something like, but feel free to ack:

```
if (AstDeclaration.isSupportedSyntaxKind(span.parent.kind)) {
  if (span.parent.kind === ts.SyntaxKind.EnumDeclaration) {
    sortChildren = collector.extractorConfig.enumMemberOrder === EnumMemberOrder.ByName;
  } else {
    sortChildren = true;
  }
}
```

Reply:
Can do, I've been generally conditioned for each branch of logic to be as simply determined, but I prefer this way of notating it.